processor/k8sattributesprocessor/options.go

```diff
@@ -201,12 +201,19 @@ func withExtractMetadata(fields ...string) option {
 func withOtelAnnotations(enabled bool) option {
 	return func(p *kubernetesprocessor) error {
 		if enabled {
-			p.rules.Annotations = append(p.rules.Annotations, kube.FieldExtractionRule{
-				Name:                 "$1",
-				KeyRegex:             regexp.MustCompile(`^resource\.opentelemetry\.io/(.+)$`),
-				HasKeyRegexReference: true,
-				From:                 kube.MetadataFromPod,
-			})
+			p.rules.Annotations = append(p.rules.Annotations, kube.OtelAnnotations())
+		}
+		return nil
+	}
+}
+
+func withServiceAttributes(config ServiceAttributeConfig) option {
+	return func(p *kubernetesprocessor) error {
+		if config.Enabled {
+			p.rules.Service = true
+			if config.Labels {
```

Comment:
Why we need this level of granularity here? Shouldn't the automatic logic of https://opentelemetry.io/docs/specs/semconv/non-normative/k8s-attributes/ just take place as is through one single setting. I suggest we remove `well_known_labels` settings if there is no strong reason for having it.

Reply:
The reason from the spec: 

> Tools offering this functionality should provide an opt-in flag for the use of well-known labels, since users may not be aware that their labels are being used for this purpose.

Reply:
Haven't noticed that, thank's! My question then is why the spec instructs this 🤔 Why only labels and not annotations too? Is there a specific reason for caring only for labels specifically?

Reply:
yes - the reason is that some/many users use those well-known labels already.

So cautious users might want to use only the annotations, as they are otel specific.
This should not prevent them from using the service attributes.

Reply:
All right, I see! We should make sure to document this so as to clearly mention its usability.

Reply:
good point - I added a note in the readme